streem/pbandk
# Publish to mavencentral

⚠️ This code must still be tested before merging ⚠️ 

Open items to do:

- [ ] Create Streem Sonatype account
- [ ] Add `SONATYPE_API_USER` and `SONATYPE_API_KEY` as Github actions secrets
- [ ] Create Streem PGP key and publish it on a public key server
- [ ] Add `SONATYPE_PGP_KEY_BASE64` as Github actions secret (export key using `gpg --export-secret-key --armor | base64`)
- [ ] Look into how to support snapshots
- [ ] Test upload to Maven Central

-------------------------------

This PR converts all existing plumbing to upload to Maven Central instead of Bintray. Key changes are:

1. Rename all Bintray/JCenter references to Sonatype/Maven Central
2. Change maven URL
3. Add artifact signing
4. Update pom.xml so it contains all required attributes for Maven Central (see here https://central.sonatype.org/publish/requirements/#sufficient-metadata)
